src/main.cpp

```diff
@@ -70,49 +83,109 @@ float normalizedToScreenY(const float n_y, const int screen_h) {
   return n_y * screen_h;
 }
 
-void drawBackground(SDL_Renderer *renderer) {
+void NormalizedElement::move(const int dx, const int dy) {
+  scr_rect.x += dx;
+  scr_rect.y += dy;
+
+  for (NormalizedConnection& connection_to_move : connections) {
+    connection_to_move.move(dx, dy);
+  }
+}
+
+void NormalizedConnection::move(const int dx, const int dy) {
+  for (SDL_FPoint& vertex_to_move : scr_vertices) {
+    vertex_to_move.x += dx;
+    vertex_to_move.y += dy;
+  }
+}
+
+void NormalizedElement::scale(
+    const float scaling_factor,
+    const int mouse_x, 
+    const int mouse_y) {
+  scr_rect.x = mouse_x + (scr_rect.x - mouse_x) * scaling_factor;
+  scr_rect.y = mouse_y + (scr_rect.y - mouse_y) * scaling_factor;
+  scr_rect.w *= scaling_factor;
+  scr_rect.h *= scaling_factor;
+
+  for (NormalizedConnection& connection_to_scale : connections) {
+    connection_to_scale.scale(scaling_factor, mouse_x, mouse_y);
+  }
+}
+
+void NormalizedConnection::scale(
+    const float scaling_factor,
+    const int mouse_x, 
+    const int mouse_y) {
+  for (SDL_FPoint& vertex_to_scale : scr_vertices) {
+    vertex_to_scale.x = mouse_x + (vertex_to_scale.x - mouse_x) * scaling_factor;
+    vertex_to_scale.y = mouse_y + (vertex_to_scale.y - mouse_y) * scaling_factor;
+  }
+}
+
+void NormalizedConnection::normalizedToScreen(
+    const int screen_w,
+    const int screen_h) {
+  for (const NormalizedPoint& n_vertex : n_vertices) {
+    SDL_FPoint vertex;
+    vertex.x = normalizedToScreenX(n_vertex.n_x, screen_w);
+    vertex.y = normalizedToScreenX(n_vertex.n_y, screen_h);
+    scr_vertices.push_back(vertex);
+  }
+}
+
+void NormalizedElement::normalizedToScreen(
+    const int screen_w,
+    const int screen_h) {
+  scr_rect.x = normalizedToScreenX(n_point.n_x, screen_w);
+  scr_rect.y = normalizedToScreenY(n_point.n_y, screen_h);
+  scr_rect.w = normalizedToScreenX(n_w, screen_w);
+  scr_rect.h = normalizedToScreenY(n_h, screen_h);
+}
+
+void drawBackground(SDL_Renderer* renderer) {
   SDL_SetRenderDrawColor(renderer, 0, 0, 0, SDL_ALPHA_OPAQUE);
   SDL_RenderClear(renderer);
 }
 
 int parseInput(
-    const char *filename,
-    std::vector<NormalizedElement> &elements_to_parse) {
+    const char* filename,
+    std::vector<NormalizedElement>& elements_to_parse) {
   pugi::xml_document file;
   if (!file.load_file(filename)) {
     return PARSER_FAILURE;
   }
 
   pugi::xml_node elements = file.child("logic_scheme").child("elements");
 
-  //Parsing elements from given file
+  // Parsing elements from given file
   for (pugi::xml_node element = elements.first_child();
-      element;
-      element = element.next_sibling()) {
+    element;
+    element = element.next_sibling()) {
```

Comment:
Double indentation for break-the-line cases, single indentation for general case: was good, now bad.